raiden-ts/src/transfers/epics.ts

```diff
@@ -476,125 +574,87 @@ export const transferGenerateAndSignEnvelopeMessageEpic = (
  *
  * @param action$ - Observable of transferSigned actions
  * @param state$ - Observable of RaidenStates
+ * @param latest$ - RaidenEpicDeps latest
  * @returns Observable of messageSend actions
  */
 export const transferSignedRetryMessageEpic = (
   action$: Observable<RaidenAction>,
   state$: Observable<RaidenState>,
+  { latest$ }: RaidenEpicDeps,
 ): Observable<ActionType<typeof messageSend>> =>
-  state$.pipe(
-    publishReplay(1, undefined, state$ =>
-      action$.pipe(
-        filter(isActionOf(transferSigned)),
-        mergeMap(action => {
-          const secrethash = action.meta.secrethash,
-            signed = action.payload.message,
-            send = messageSend({ message: signed }, { address: signed.recipient });
-          // emit Send once immediatelly, then wait until respective messageSent, then completes
-          const sendOnceAndWaitSent$ = merge(
-            of(send),
-            action$.pipe(
-              filter(
-                a =>
-                  isActionOf(messageSent, a) &&
-                  a.payload.message === send.payload.message &&
-                  a.meta.address === send.meta.address,
-              ),
-              take(1),
-              // don't output messageSent, just wait for it before completing
-              ignoreElements(),
-            ),
-          );
-          return sendOnceAndWaitSent$.pipe(
-            // Resubscribe/retry every 30s after messageSend succeeds with messageSent
-            // Notice first (or any) messageSend can wait for a long time before succeeding, as it
-            // waits for address's user in transport to be online and joined room before actually
-            // sending the message. That's why repeatWhen emits/resubscribe only some time after
-            // sendOnceAndWaitSent$ completes, instead of a plain 'interval'
-            // TODO: configurable retry delay, possibly use an exponential backoff timeout strat
-            repeatWhen(completed$ => completed$.pipe(delay(30e3))),
-            // until transferProcessed received OR transfer completed OR channelClosed
-            takeUntil(
-              state$.pipe(
-                filter(
-                  state =>
-                    !!state.sent[secrethash].transferProcessed ||
-                    !!state.sent[secrethash].unlockProcessed ||
-                    !!state.sent[secrethash].lockExpiredProcessed ||
-                    !!state.sent[secrethash].channelClosed,
-                ),
-              ),
-            ),
-          );
-        }),
-      ),
+  action$.pipe(
+    filter(isActionOf(transferSigned)),
+    mergeMap(action =>
+      transferSignedRetryMessage(action$, latest$.pipe(pluckDistinct('state')), action),
     ),
   );
 
+const transferUnlockRetryMessage$ = (
+  action$: Observable<RaidenAction>,
+  state$: Observable<RaidenState>,
+  action: ActionType<typeof transferUnlocked>,
+  state: RaidenState,
+) => {
```

Comment:
Would be great to have a descriptive return type

Reply:
I am not against that but it would be better to enforce return types if we want to be consistent.